pkg/reconciler/topic/topic.go

```diff
@@ -0,0 +1,420 @@
+			state, err := c.EnsureTopicDeleted(ctx, topic, topic.Spec.Project, topic.Status.TopicID)
+			switch state {
+			case pubsub.OpsJobGetFailed:
+				logger.Error("Failed to get Topic ops job.", zap.Any("state", state), zap.Error(err))
+				return err
+
+			case pubsub.OpsJobCreated:
+				// If we created a job to make a subscription, then add the finalizer and update the status.
+				topic.Status.MarkTopicOperating(
+					"Deleting",
+					"Created Job to delete topic %q.",
+					topic.Status.TopicID)
+				return nil
+
+			case pubsub.OpsJobCompleteSuccessful:
+				topic.Status.MarkNoTopic("Deleted", "Successfully deleted topic %q.", topic.Status.TopicID)
+				topic.Status.TopicID = ""
+				removeFinalizer(topic)
+
+			case pubsub.OpsJobCreateFailed, pubsub.OpsJobCompleteFailed:
+				logger.Error("Failed to delete topic.", zap.Any("state", state), zap.Error(err))
+
+				msg := "unknown"
+				if err != nil {
+					msg = err.Error()
+				}
+				topic.Status.MarkNoTopic(
+					"DeleteFailed",
+					"Failed to delete topic: %q.",
+					msg)
+				return err
+			}
+
+		default:
+			removeFinalizer(topic)
+		}
+
+		return nil
+	}
+
+	// Set the topic being used.
+	topic.Status.TopicID = topic.Spec.Topic
+
+	switch topic.Spec.PropagationPolicy {
+	case v1alpha1.TopicPolicyCreateDelete, v1alpha1.TopicPolicyCreateNoDelete:
+		state, err := c.EnsureTopicCreated(ctx, topic, topic.Spec.Project, topic.Status.TopicID)
+		// Check state.
+		switch state {
+		case pubsub.OpsJobGetFailed:
+			logger.Error("Failed to get topic ops job.",
+				zap.Any("propagationPolicy", topic.Spec.PropagationPolicy),
+				zap.Any("state", state),
+				zap.Error(err))
+			return err
+
+		case pubsub.OpsJobCreated:
+			// If we created a job to make a subscription, then add the finalizer and update the status.
+			addFinalizer(topic)
+			topic.Status.MarkTopicOperating("Creating",
+				"Created Job to create topic %q.",
+				topic.Status.TopicID)
+			return nil
+
+		case pubsub.OpsJobCompleteSuccessful:
+			topic.Status.MarkTopicReady()
+
+		case pubsub.OpsJobCreateFailed, pubsub.OpsJobCompleteFailed:
+			logger.Error("Failed to create topic.",
+				zap.Any("propagationPolicy", topic.Spec.PropagationPolicy),
+				zap.Any("state", state),
+				zap.Error(err))
+
+			msg := "unknown"
+			if err != nil {
+				msg = err.Error()
+			}
+			topic.Status.MarkNoTopic(
+				"CreateFailed",
+				"Failed to create Topic: %q",
+				msg)
+			return err
+		}
+
+	case v1alpha1.TopicPolicyNoCreateNoDelete:
+		state, err := c.EnsureTopicExists(ctx, topic, topic.Spec.Project, topic.Status.TopicID)
+		// Check state.
+		switch state {
+		case pubsub.OpsJobGetFailed:
+			logger.Error("Failed to get topic ops job.",
+				zap.Any("propagationPolicy", topic.Spec.PropagationPolicy),
+				zap.Any("state", state),
+				zap.Error(err))
+			return err
+
+		case pubsub.OpsJobCreated:
+			// If we created a job to make a subscription, then add the finalizer and update the status.
```

Comment:
This comment seems inaccurate... doesn't this job just verify the topic exists? Or does it create a subscription too?

Reply:
yeah, updated, thanks